fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/centralstore/CentralStore.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * CentralStore.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.centralstore;
+
+import com.apple.foundationdb.record.RecordMetaData;
+import com.apple.foundationdb.record.RecordMetaDataBuilder;
+import com.apple.foundationdb.record.RecordMetaDataProto;
+import com.apple.foundationdb.record.metadata.Key;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContext;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStore;
+import com.apple.foundationdb.record.provider.foundationdb.MetaDataProtoEditor;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.function.Function;
+
+/**
+ * TODO: CentralStore
+ */
+public class CentralStore {
```

Comment:
As elucidated below, maybe this should be something more like `CentralStoreFactory`, in which you provide a `KeySpacePath` that identifies where it is along with an `FDBRecordContext` and it returns an instance of a `CentralStore`.  I think, also, the `CentralStore` should internally wrap an instance of the record store itself and present API's to the rest of the record layer to do things like update the metadata about an indexes state, e.g.:

    centralStore.setIndexState(Index index, IndexState state);

(probably an oversimplification here, but you get the idea). 

The main point being that the central store has a clean API for all of the users that the record layer internally needs to maintain metadata about record stores, then if an outside application needs to do more, it can ask for the underlying record store.